packages/serverpod/lib/src/database/database_connection.dart

```diff
@@ -295,26 +295,23 @@ class DatabaseConnection {
   }) async {
     var startTime = DateTime.now();
 
-    var data = row.toJsonForDatabase();
+    var selectedColumns = columns ?? row.table.columns;
+    Map data = row.allToJson();
 
-    columns?.forEach((column) {
-      if (!data.containsKey(column.columnName)) {
-        throw Exception(
-          'Column ${column.columnName} does not exist in table row provided.',
-        );
-      }
-    });
-
-    int? id = data['id'];
-
-    Iterable<String> cols = columns?.map((c) => c.columnName) ?? data.keys;
+    for (var column in selectedColumns) {
+      assert(data.containsKey(column.columnName));
+    }
 
-    String updates = cols.where((column) => column != 'id').map((column) {
-      var value = DatabasePoolManager.encoder.convert(data[column]);
-      return '"$column" = $value';
+    var updates = selectedColumns
+        .where((column) => column.columnName != 'id')
+        .map((column) {
+      var value = DatabasePoolManager.encoder.convert(data[column.columnName]);
+      return '"${column.columnName}" = $value';
     }).join(', ');
 
-    var query = 'UPDATE "${row.tableName}" SET $updates WHERE id = $id';
+    int? id = data['id'];
```

Comment:
Do we have a check to see if `id` is null somewhere? Or how do we behave in that case?

Reply:
I don't think we do, I believe it would just crash on an SQL error, I can add an explicit check for this!

Reply:
Might as well add an exception for it with a helpful error message that can help the developer understand that the ID is missing.